misk-api/src/main/kotlin/misk/api/RequestContext.kt

```diff
@@ -0,0 +1,16 @@
+package misk.api
+
+import misk.web.DispatchMechanism
+import okhttp3.HttpUrl
+
+interface RequestContext {
+  val url: HttpUrl
+  val dispatchMechanism: DispatchMechanism
+  val requestHeadersContext: RequestHeadersContext
+}
+
+interface RequestHeadersContext {
+  operator fun get(name: String): String?
+
+  fun asIterable(): Iterable<Pair<String, String>>
```

Comment:
The Okhttp Headers class implements the `Iterable<Pair<String, String>>` interface.